test/built-ins/Temporal/Duration/from/string.js

```diff
@@ -0,0 +1,179 @@
+    -1, -2, -3, -4, -5, -6, -7, -8, -9, -1, -1, false);
+// positive sign
+assertDuration(Temporal.Duration.from("+P1Y2M3W4DT5H6M7.008009001S"),
+    1, 2, 3, 4, 5, 6, 7, 8, 9, 1, 1, false);
+
+assertDuration(Temporal.Duration.from("PT2.5H"),
+    0, 0, 0, 0, 2, 30, 0, 0, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT2.25H"),
+    0, 0, 0, 0, 2, 15, 0, 0, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT2.05H"),
+    0, 0, 0, 0, 2, 3, 0, 0, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT2.005H"),
+    0, 0, 0, 0, 2, 0, 18, 0, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT2.505H"),
+    0, 0, 0, 0, 2, 30, 18, 0, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT2.0025H"),
+    0, 0, 0, 0, 2, 0, 9, 0, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3.5M"),
+    0, 0, 0, 0, 0, 3, 30, 0, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3.25M"),
+    0, 0, 0, 0, 0, 3, 15, 0, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3.125M"),
+    0, 0, 0, 0, 0, 3, 7, 500, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3.025M"),
+    0, 0, 0, 0, 0, 3, 1, 500, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3.01M"),
+    0, 0, 0, 0, 0, 3, 0, 600, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3.005M"),
+    0, 0, 0, 0, 0, 3, 0, 300, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3.001M"),
+    0, 0, 0, 0, 0, 3, 0, 60, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3.006M"),
+    0, 0, 0, 0, 0, 3, 0, 360, 0, 0, 1, false);
+
+// Use , instead of .
+assertDuration(Temporal.Duration.from("PT2,5H"),
+    0, 0, 0, 0, 2, 30, 0, 0, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT2,25H"),
+    0, 0, 0, 0, 2, 15, 0, 0, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT2,05H"),
+    0, 0, 0, 0, 2, 3, 0, 0, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT2,005H"),
+    0, 0, 0, 0, 2, 0, 18, 0, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT2,505H"),
+    0, 0, 0, 0, 2, 30, 18, 0, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT2,0025H"),
+    0, 0, 0, 0, 2, 0, 9, 0, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3,5M"),
+    0, 0, 0, 0, 0, 3, 30, 0, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3,25M"),
+    0, 0, 0, 0, 0, 3, 15, 0, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3,125M"),
+    0, 0, 0, 0, 0, 3, 7, 500, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3,025M"),
+    0, 0, 0, 0, 0, 3, 1, 500, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3,01M"),
+    0, 0, 0, 0, 0, 3, 0, 600, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3,005M"),
+    0, 0, 0, 0, 0, 3, 0, 300, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3,001M"),
+    0, 0, 0, 0, 0, 3, 0, 60, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3,006M"),
+    0, 0, 0, 0, 0, 3, 0, 360, 0, 0, 1, false);
```

Comment:
What happens when the wrong number of arguments are passed?

Reply:
"PT3,001M" in ISO8601 is a different way to represent "PT3.001M" , whic means 3.001 minutes.
which is 3 minutes and 60 milliseconds

"PT3,006M" in ISO8601 is a different way to represent "PT3.006M" , whic means 3.006 minutes.
which is 3 minutes and 360 milliseconds

See https://tc39.es/proposal-temporal/#prod-DecimalSeparator both "." and "," are DecimalSeparator .

If your question is about the number of argument of the from method- there is only one 
https://tc39.es/proposal-temporal/#sec-temporal.duration.from